Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/MultilineParametersConfiguration.swift

```diff
@@ -8,4 +8,18 @@ struct MultilineParametersConfiguration: SeverityBasedRuleConfiguration {
     private(set) var severityConfiguration = SeverityConfiguration<Parent>(.warning)
     @ConfigurationElement(key: "allows_single_line")
     private(set) var allowsSingleLine = true
+    @ConfigurationElement(key: "max_number_of_single_line_parameters")
+    private(set) var maxNumberOfSingleLineParameters: Int?
+
+    func validate() throws {
+        if let maxNumberOfSingleLineParameters, !allowsSingleLine && maxNumberOfSingleLineParameters > 1 {
+            throw Issue.invalidConfiguration(
+                ruleID: Parent.identifier,
+                message: """
+                         Invalid configuration: 'allows_single_line' and 'max_number_of_single_line_parameters' are mutually exclusive.
+                         They cannot both be active with conflicting values.
+                         """
+            )
+        }
```

Comment:
Note that the `Invalid configuration: ` part will be added to the message by default.

We could also directly check, that `max_number_of_single_line_parameters` has a sane value in any case.

My suggestion:

```suggestion
        guard let maxNumberOfSingleLineParameters else {
            return
        }
        guard maxNumberOfSingleLineParameters >= 1 else {
            throw Issue.invalidConfiguration(
                ruleID: Parent.identifier,
                message: "Option '\($maxNumberOfSingleLineParameters.key)' should be >= 1."
            )
        }
        if maxNumberOfSingleLineParameters > 1, !allowsSingleLine {
            throw Issue.inconsistentConfiguration(
                ruleID: Parent.identifier,
                message: """
                    Option '\($maxNumberOfSingleLineParameters.key)' has no effect when \
                    '\($allowsSingleLine.key)' is false.
                    """
            )
        }
```